src/chainstate/coordinator/mod.rs

```diff
@@ -196,7 +217,32 @@ impl<'a, T: BlockEventDispatcher>
             stacks_chain_id,
             chain_state_path,
             initial_balances,
-            boot_block_exec,
+            |clarity_tx| {
+                let burnchain = burnchain.clone();
+                let contract = QualifiedContractIdentifier::parse(&format!(
+                    "{}.pox",
+                    STACKS_BOOT_CODE_CONTRACT_ADDRESS
+                ))
+                .expect("Failed to construct boot code contract address");
+                let sender = PrincipalData::from(contract.clone());
+
+                clarity_tx.connection().as_transaction(|conn| {
```

Comment:
This is going to fail when run on mainnet, because the `set-burnchain-parameters` method is wired to only run if the node is on regtest/testnet.  This is because this method has this line in it:

```
        (asserts! (and is-in-regtest (not (var-get configured))) (err ERR_NOT_ALLOWED))
```

Reply:
That check was removed in this PR.

Reply:
Nevermind -- I see that you fixed that already.